src/transaction/log_replication_atomic.cpp

```diff
@@ -233,20 +234,20 @@ namespace cublog
 	      }
 	    break;
 	  }
-	  case LOG_SCHEMA_MODIFICATION_LOCK:
+	  case LOG_LOCKED_OBJECT:
 	  {
-	    m_redo_context.m_reader.advance_when_does_not_fit (sizeof (LOG_REC_SCHEMA_MODIFICATION_LOCK));
-	    const LOG_REC_SCHEMA_MODIFICATION_LOCK log_rec =
-		    m_redo_context.m_reader.reinterpret_copy_and_add_align<LOG_REC_SCHEMA_MODIFICATION_LOCK> ();
+	    m_redo_context.m_reader.advance_when_does_not_fit (sizeof (LOG_REC_LOCKED_OBJECT));
+	    const LOG_REC_LOCKED_OBJECT log_rec =
+		    m_redo_context.m_reader.reinterpret_copy_and_add_align<LOG_REC_LOCKED_OBJECT> ();
 
 	    if (is_locked_for_ddl (header.trid, &log_rec.classoid))
```

Comment:
```suggestion
	    if (is_locked_for_ddl (header.trid, &log_rec.oid))
```
Am I correct? it seems to check Rootclass and db_serial.. Then I don't know how it passed your tests. Sorry if I'm missing something.

Reply:
!!! Oh... thanks..

it's possible that the lock record was applied multiple times. While there may not be any functional issues, the `m_locked_object` data structure could potentially contain multiple instances of the same lock information. It seems I overlooked this while requesting and releasing the same lock multiple times. Since there don't seem to be any functional problems, I'll fix it. Thank you.